pkg/operators/clustermanager/controllers/migrationcontroller/migration_controller.go

```diff
@@ -103,27 +113,33 @@ func (c *crdMigrationController) sync(ctx context.Context, controllerContext fac
 	if err != nil {
 		return err
 	}
-	apiExtensionClient, err := apiextensionsclient.NewForConfig(hubKubeconfig)
+	apiExtensionClient, migrationClient, err := c.generateHubClusterClients(hubKubeconfig)
 	if err != nil {
 		return err
 	}
-	migrationClient, err := migrationv1alpha1client.NewForConfig(hubKubeconfig)
-	if err != nil {
-		return err
+
+	// ClusterManager is deleting, we remove its related resources on hub
+	if !clusterManager.DeletionTimestamp.IsZero() {
+		return removeStorageVersionMigrations(ctx, migrationClient)
 	}
 
 	// apply storage version migrations if it is supported
 	supported, err := supportStorageVersionMigration(ctx, apiExtensionClient)
 	if err != nil {
 		return err
 	}
-	if !supported {
-		return nil
-	}
 
-	// ClusterManager is deleting, we remove its related resources on hub
-	if !clusterManager.DeletionTimestamp.IsZero() {
-		return removeStorageVersionMigrations(ctx, migrationClient)
+	if !supported {
+		migrationCond := metav1.Condition{
+			Type:    MigrationSucceeded,
+			Status:  metav1.ConditionFalse,
+			Reason:  "StorageVersionMigrationFailed",
+			Message: fmt.Sprintf("Do not support StorageVersionMigration"),
+		}
+		_, _, err = helpers.UpdateClusterManagerStatus(ctx, c.clusterManagerClient, clusterManagerName,
+			helpers.UpdateClusterManagerConditionFn(migrationCond),
+		)
+		return err
```

Comment:
Does this mean if not supported, CRD stored version will never be removed?

Reply:
Yes, user need to manually delete it or add a `MigrationSucceeded` condition in clustermanger status condition.